src/db.cpp

```diff
@@ -211,7 +211,7 @@ bool CDBEnv::Salvage(std::string strFile, bool fAggressive,
     while (!strDump.eof() && keyHex != "DATA=END")
     {
         getline(strDump, keyHex);
-        if (keyHex != "DATA_END")
+        if (keyHex != "DATA=END")
```

Comment:
doh! I can see where this bug came from. Other satoshi derivs use DATA_END as a var defined as string "DATA=END"